mantis-runtime-loader/src/main/java/io/mantisrx/runtime/loader/config/WorkerConfiguration.java

```diff
@@ -192,4 +193,16 @@ default Map<String, String> getTaskExecutorAttributes() {
             .filter(entry -> !entry.getValue().matches("\\$\\{.*\\}"))
             .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
     }
+
+    // ------------------------------------------------------------------------
+    //  Routing related configurations
+    // ------------------------------------------------------------------------
+    @Config("mantis.taskexecutor.router.scalar-stage-to-stage")
+    @Default("io.reactivex.mantis.network.push.RoundRobinRouter")
+    String getScalarStageToStageRouterClassName();
+
+
+    @Config("mantis.availabilityZoneUtils.class")
+    @Default("io.mantisrx.common.util.DefaultAvailabilityZoneUtils")
+    AvailabilityZoneUtils getAvailabilityZoneUtils();
```

Comment:
please avoid non-primitive types in the worker config as they've been causing many trouble for us in integration and making migration to other config framework harder.
I would recommend using either ctor injection, serviceLocator or reconstruct using primitive types here.